soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -161,6 +164,7 @@ impl TokenTrait for Token {
             ClassicMetadata::AlphaNum12(asset) => {
                 //TODO: Better way to do this?
                 let mut code12 = [0u8; 12];
+                e.charge_budget(CostType::BytesClone, 12)?;
```

Comment:
Is there a reason we can't use the len of the value we're measuring?
```suggestion
                e.charge_budget(CostType::BytesClone, code12.len())?;
```